modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-impl/src/main/java/com/liferay/headless/admin/taxonomy/internal/resource/v1_0/TaxonomyCategoryResourceImpl.java

```diff
@@ -575,35 +572,48 @@ private String[] _merge(
 		List<AssetCategoryProperty> assetCategoryProperties,
 		TaxonomyCategoryProperty[] taxonomyCategoryProperties) {
 
-		Stream<TaxonomyCategoryProperty> stream = Arrays.stream(
-			Optional.ofNullable(
-				taxonomyCategoryProperties
-			).orElse(
-				new TaxonomyCategoryProperty[0]
-			));
+		if (ArrayUtil.isEmpty(taxonomyCategoryProperties)) {
+			String[] categoryProperties =
+				new String[assetCategoryProperties.size()];
 
-		Map<String, String> map = stream.collect(
-			Collectors.toMap(
-				TaxonomyCategoryProperty::getKey,
-				TaxonomyCategoryProperty::getValue));
+			for (int i = 0; i < categoryProperties.length; i++) {
```

Comment:
@slnn This part basically transforms a list to a string array, please try to use TransformUtil.

Reply:
@slnn Actually this short cut may not be needed, or is only needed when the taxonomyCategoryProperties array is actually `null`.

Reply:
If you just do 
```
if (taxonomyCategoryProperties == null) {
    taxonomyCategoryProperties = new ...;
}
```
it's totally fine, too.

Reply:
Hi @dantewang 

In the original logic, when the array taxonomyCategoryProperties is null or empty, then the `stream.collect` will not return anything to the map, it just needs to handle the list assetCategoryProperties, so I add the shortcut here, do I make any mistakes about the original logic?

```
Stream<TaxonomyCategoryProperty> stream = Arrays.stream(
			Optional.ofNullable(
				taxonomyCategoryProperties
			).orElse(
				new TaxonomyCategoryProperty[0]
			));

		Map<String, String> map = stream.collect(
			Collectors.toMap(
				TaxonomyCategoryProperty::getKey,
				TaxonomyCategoryProperty::getValue));
```


Lily

Reply:
Basically I don't want to see you write the same loop twice.

Reply:
OK, I see